core/src/main/java/com/rudderstack/android/sdk/core/RudderDeviceModeManager.java

```diff
@@ -242,8 +243,10 @@ private void replayMessageQueue() {
             RudderLogger.logDebug(String.format(Locale.US, "RudderDeviceModeManager: replayMessageQueue: replaying old messages with factories. Count: %d", messageIds.size()));
             for (int i = 0; i < messageIds.size(); i++) {
                 try {
-                    RudderMessage message = RudderGson.getInstance().fromJson(messages.get(i), RudderMessage.class);
-                    makeFactoryDump(message, messageIds.get(i), true);
+                    RudderMessage message = RudderGson.deserialize(messages.get(i), RudderMessage.class);
+                    if (message != null) {
+                        makeFactoryDump(message, messageIds.get(i), true);
```

Comment:
Maybe this methods needs some renaming, as `dump` to `processMessage` was renamed a bit above. What do you think?

Reply:
Agreed, renamed `dump` as appropriate across the code.